src/beast/beast/http/string_body.h

```diff
@@ -0,0 +1,116 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of Beast: https://github.com/vinniefalco/Beast
+    Copyright 2013, Vinnie Falco <vinnie.falco@gmail.com>
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef BEAST_HTTP_STRING_BODY_H_INCLUDED
+#define BEAST_HTTP_STRING_BODY_H_INCLUDED
+
+#include <beast/http/error.h>
+#include <beast/http/message.h>
+#include <beast/asio/append_buffers.h>
+#include <beast/asio/streambuf.h>
+#include <memory>
+#include <string>
+
+namespace beast {
+namespace http {
+
+/** A Body represented by a std::string.
+*/
+struct string_body
+{
+    using value_type = std::string;
+
+    static bool constexpr is_simple = true;
```

Comment:
This is changed in a later PR, it becomes `is_single_pass`, actually gets used, and all the bodies set it. But even this is subject to change.